lib/sequel/plugins/enum_values.rb

```diff
@@ -68,20 +70,30 @@ def enum_values(field)
 
 				private
 
-				def transform_predicate_methods_to_enum_fields(predicate_methods)
-					case predicate_methods
-					when TrueClass
-						all_enum_fields.keys
-					when FalseClass
-						[]
-					else
-						Array(predicate_methods)
+				def process_enum_values_options
+					@enum_values_caching =
+						@enum_values_options.fetch(:caching, @enum_values_caching)
+
+					process_enum_values_option(:filter_methods) do |field, value|
+						dataset.methods.include?(value.to_sym) &&
+							warn("WARNING: Redefining method #{value}")
```

Comment:
1. Approach with `&&` for condition very unclear, please use `if`/`unless`.
2. I see no reason for parentheses of `warn`.
3. It's a pretty good idea to warn, but I suggest to make it even strict: let raise an exception! And see what will happens with users.

Reply:
4. And there are should be tests for this behavior!